### PR TITLE
Test that change password link exists: #568

### PR DESCRIPTION
**Pull request by cecilia-donnelly**
PR #568 fixed an issue where the change password link wasn't showing up
for providers.  This tests that, and will fail if the link disappears
again.  This also adds ignored features to be implemented later, testing
whether the change password link functions correctly.

@jasonaowen, I added ids to these elements even after our discussion in #585.  These links don't have existing classes, and finding them in the DOM was difficult.  Happy to discuss or remove if you think this isn't the right way to do it.  

Also, this should be merged after #585 so I can easily rebase/edit this to use the new step library introduced there.  Finally, if reviewers would rather not have `@ignore`'d features here then I can take them out or implement them as desired.

Thanks!  Testing this should just be running the integration tests, possibly editing/removing the change password link to see the test fail.

### CODE REVIEW

**Comment by jasonaowen** [comment]
Now that #585 has been merged, would you mind rebasing this onto `master`, @cecilia-donnelly ?

**Comment by cecilia-donnelly** [comment]
Just to give you an update: I now have a duplicate definition of the login method for Change Password and Logout tests, so I need to resolve that.

**Comment by cecilia-donnelly** [comment]
Rebased and updated the test.  Ready for review!

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/general/steps/ChangePasswordStepDefinitions.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import cucumber.api.java.en.When;
+import net.thucydides.core.annotations.Steps;
+
+import gov.medicaid.features.general.ui.DashboardPage;
+import gov.medicaid.features.general.ui.MyProfilePage;
+
+@SuppressWarnings("unused")
+public class ChangePasswordStepDefinitions {
+    @Steps
+    GeneralSteps generalSteps;
```
Comment (jasonaowen) [inline psm-app/integration-tests/src/test/java/gov/medicaid/features/general/steps/ChangePasswordStepDefinitions.java]: This variable appears to be unused; please delete.

**Comment by jasonaowen** [comment]
There's a larger issue that I wanted to put in a regular conversation instead of in a review.

One of the features Serenity offers is to take screenshots of the page at each step. In order to trigger that, the step definitions class (with `@Given`, `@When`, `@Then` annotations) needs to call methods that are annotated with `@Step`. The pattern that I think I learned from @BenGalewsky seems to be: a step definition class has a member variable annotated with `@Steps`, the methods step definition classes call are annotated with `@Step`, and the steps class holds on to the `...Page` objects.

I'm not sure how useful that separation is, and I know we haven't been perfect about following it, but it makes sense to me as someone who has not written very much Serenity / Selenium.

What I suggest, then, is to add methods to the `GeneralSteps` class, remove the page objects from `ChangePasswordStepDefinitions`, and call into `GeneralSteps` instead. (Which means that you should ignore [my review comment](https://github.com/SolutionGuidance/psm/pull/591#discussion_r159516819).

Does that make sense?

File: psm-app/integration-tests/src/test/java/gov/medicaid/features/general/steps/ChangePasswordStepDefinitions.java (hunk 1 above)
Comment (jasonaowen) [reply psm-app/integration-tests/src/test/java/gov/medicaid/features/general/steps/ChangePasswordStepDefinitions.java]: [Wait, nevermind](https://github.com/SolutionGuidance/psm/pull/591#issuecomment-355126975).

**Comment by cecilia-donnelly** [comment]
Hmm, I didn't get any errors with `checkstyleMain`, @jasonaowen.  Was there something in particular you saw?

**Comment by jasonaowen** [comment]
@cecilia-donnelly `checkstyleMain` only runs checkstyle against files in `src/main`; you'd need to run `checkstyleTest` to find checkstyle errors in files in `src/test`. See also the [Travis build log](https://travis-ci.org/SolutionGuidance/psm/builds/325181591#L835).

**Comment by cecilia-donnelly** [comment]
@jasonaowen ooooooh, thank you!

**Comment by cecilia-donnelly** [comment]
Moved the methods into the `Step` library and `checkstyleTest` now passes.  Ready for re-review, @jasonaowen.

**Comment by cecilia-donnelly** [comment]
@jasonaowen I want to at least keep the last commit (a442c90) separate, since it includes some information about how Serenity works that I know I'll be looking back at later.  I can certainly squash the first two (46f5dc9 and f290976) together, though!  Thanks for the review. :)

**Pull request merged**